src/simple_model.jl

```diff
@@ -128,7 +128,7 @@ end
 
 function jprod!(nlp :: SimpleNLPModel, x :: Vector, v :: Vector, Jv :: Vector)
   nlp.counters.neval_jprod += 1
-  Jv[:] = ForwardDiff.jacobian(nlp.c, x) * v
+  Jv[1:nlp.meta.ncon] = ForwardDiff.jacobian(nlp.c, x) * v
```

Comment:
Speedwise is the same, and this (should) break less, so maybe this way is better.
If we change this back, we have to change Ampl back too.